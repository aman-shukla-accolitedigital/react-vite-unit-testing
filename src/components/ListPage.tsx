import React, { useState } from 'react';

const ListPage: React.FC = () => {
  const [items, setItems] = useState<string[]>(['Item 1', 'Item 2', 'Item 3']);
  const [newItem, setNewItem] = useState<string>('');

  const addItem = () => {
    if (newItem.trim()) {
      setItems([...items, newItem]);
      setNewItem('');
    }
  };

  return (
    <div>
      <h1>List Page</h1>
      <input
        type="text"
        value={newItem}
        onChange={(e) => setNewItem(e.target.value)}
        placeholder="Add new item"
      />
      <button onClick={addItem}>Add Item</button>
      <ul>
        {items.map((item, index) => (
          <li key={index}>{item}</li>
        ))}
      </ul>
    </div>
  );
};

export default ListPage;