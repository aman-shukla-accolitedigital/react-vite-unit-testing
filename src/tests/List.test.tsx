import { describe, it, expect } from 'vitest'
import { fireEvent, render, screen } from '@testing-library/react'
import ListPage from '../components/ListPage';
describe('About Component', () => {

  it('renders the About component', () => {
    const { asFragment} = render(
      <ListPage />)
        // Find the input element
      expect(asFragment()).toMatchSnapshot();
      const inputElement = screen.getByRole('textbox');

      // Simulate typing into the input field
      fireEvent.change(inputElement, { target: { value: 'Item 4' } });
      fireEvent.click(screen.getByRole('button', { name: /add item/i }));
      expect(screen.getByText('Item 4')).toBeInTheDocument();
  })

})