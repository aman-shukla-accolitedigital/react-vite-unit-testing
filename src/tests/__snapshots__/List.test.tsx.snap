// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About Component > renders the About component 1`] = `
<DocumentFragment>
  <div>
    <h1>
      List Page
    </h1>
    <input
      placeholder="Add new item"
      type="text"
      value=""
    />
    <button>
      Add Item
    </button>
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
      <li>
        Item 3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
